compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -134,8 +133,10 @@ instance Hashable Key where
 --   not propagate diagnostic errors through multiple phases.
 type IdeResult v = ([Diagnostic], Maybe v)
 
+type IdeReturn v = Maybe v
```

Comment:
This seems like a pretty confusing type synonym, especially given the fact that we already have `IdeResult`. I would just inline this.

Reply:
I think this is something that you'd reasonably want to change at a later date and it's all over the place

Reply:
Tat would be true if we hadn't just introduced IdeReturn but reused IdeResult. Since we are changing it everywhere, might as well change it to the end result.